spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/factory/HystrixGatewayFilterFactory.java

```diff
@@ -100,11 +102,36 @@ public GatewayFilter apply(Config config) {
 				s.onCancel(sub::unsubscribe);
 			}).onErrorResume((Function<Throwable, Mono<Void>>) throwable -> {
 				if (throwable instanceof HystrixRuntimeException) {
+					/*
+					 * Special handling of error events that need to be mapped to specific status codes,
+					 * instead of a generic 500
+					 */
 					HystrixRuntimeException e = (HystrixRuntimeException) throwable;
-					if (e.getFailureType() == TIMEOUT) {
-						return Mono.error(new ResponseStatusException(HttpStatus.GATEWAY_TIMEOUT));
+
+					switch (e.getFailureType()) {
+
+						case TIMEOUT:
+							return Mono.error(new ResponseStatusException(HttpStatus.GATEWAY_TIMEOUT));
+
+						case COMMAND_EXCEPTION:
+							Throwable cause = e.getCause();
+
+							if (cause instanceof ResponseStatusException) {
+								/*
```

Comment:
When https://github.com/spring-projects/spring-boot/issues/14742 is fixed upstream, here we should add support for `@ResponseStatus` using a logic similar to the one of [WebFluxResponseStatusExceptionHandler](https://github.com/spring-projects/spring-framework/blob/8aa6e5bfea2c7314deaa1b432554e9e914b09ee7/spring-webflux/src/main/java/org/springframework/web/reactive/handler/WebFluxResponseStatusExceptionHandler.java)

Reply:
Fundamentally, the condition would change to:
`cause instanceof ResponseStatusException || AnnotatedElementUtils.hasAnnotation(cause.getClass(), ResponseStatus.class)`